uber/model_checks.py

```diff
@@ -53,29 +57,48 @@ def group_paid(group):
 
 def _invalid_phone_number(s):
     if not s.startswith('+'):
-        return len(re.findall(r'\d', s)) != 10 or re.search(SAME_NUMBER_REPEATED, re.sub(r'[^0-9]','',s))
+        return len(re.findall(r'\d', s)) != 10 or re.search(c.SAME_NUMBER_REPEATED, re.sub(r'[^0-9]', '', s))
+
 
 def _invalid_zip_code(s):
     return len(re.findall(r'\d', s)) not in [5, 9]
 
-def attendee_misc(attendee):
-    if attendee.group_id and not attendee.first_name.strip() and not attendee.last_name.strip():
-        return
-
-    if COLLECT_EXACT_BIRTHDATE and attendee.birthdate == '':
-        attendee.birthdate = None # Prevent insertion errors for placeholder attendees
 
-    if not attendee.first_name or not attendee.last_name:
-        return 'First Name and Last Name are required'
-    elif attendee.placeholder:
-        return
-
-    if COLLECT_EXACT_BIRTHDATE and attendee.birthdate is None:
-        return 'Enter your date of birth.'
-    if COLLECT_EXACT_BIRTHDATE and attendee.birthdate > date.today():
-        return 'You cannot be born in the future.'
-
-    if COLLECT_FULL_ADDRESS:
+def ignore_unassigned_and_placeholders(func):
+    @wraps(func)
+    def with_skipping(attendee):
+        unassigned_group_reg = attendee.group_id and not attendee.first_name and not attendee.last_name
+        valid_placeholder = attendee.placeholder and attendee.first_name and attendee.last_name
+        if not unassigned_group_reg and not valid_placeholder:
+            return func(attendee)
+    return with_skipping
+
+
```

Comment:
I broke most of the validations out into individual methods.  This allows plugins to turn off specific validations, which they can do by name (i.e. by the function name).  We're not currently making use of that for MAGStock but some event will probably want to at some point.